### DISCUSSION TITLE
Question on using the `requires` and `config` keywords

### DISCUSSION THREAD

**Discussion by vendion** [wbthomason/packer.nvim#998] [Q&A] [answered]
I'm still trying to figure out how to use `packer`, but I like you can specify dependencies via the `requires` keyword and have it load in configuration for the plugin via `config`. What I am trying to do is have all the plugin settings in _$HOME/.config/nvim/lua/vendion/config/(file).lua_ and in my packer file I just `require` it like so:

```
use {'wbthomason/packer.nvim'}

-- Themes & Display
use {'arcticicestudio/nord-vim'}
use {
  "nvim-lualine/lualine.nvim",
  config = require('vendion.config.lualine'),
  requires = {
    { "kyazdani42/nvim-web-devicons" },
    { "SmiteshP/nvim-navic" },
  },
}
use {
  'goolord/alpha-nvim',
  requires = {
    { "kyazdani42/nvim-web-devicons" },
  },
  config = require('vendion.config.alpha-nvim')
}

-- Utilities
use {
  'TimUntersberger/neogit',
  config = require('vendion.config.neogit')
  requires = {
    {'nvim-lua/plenary.nvim'},
    {
      'sindrets/diffview.nvim',
      cmd = {
        "DiffviewOpen",
        "DiffviewClose",
        "DiffviewToggleFiles",
        "DiffviewFocusFiles",
      },
      config = require('vendion.config.diffview'),
    },
  },
  cmd = "Neogit",
}
```

This seems to be somewhat working as I was able to do `PackerInstall` and `PackerSync` and the plugins got installed. The problem comes in when `packer` tries to build the _packer\_compiled.lua_ file as it throws the following error:

```
[ERROR Wed 03 Aug 2022 04:17:04 PM -03 22173591714491] .../site/pack/packer/start/packer.nvim/lua/packer/async.lua:20: Error in coroutine: ...e/nvim/site/pack/packer/start/packer.nvim/lua/packer.lua:709: invalid value (boolean) at index 16 in table for 'concat'
```

So, what I'm wondering is if my approach is wrong or if there is a better way to do what I'm wanting or if this is a bug?

**Comment by EdenEast** [comment, marked as answer]
The issue is that `config` is expecting a function but is given what ever is returned from the lua modules that you are requiring. This is simply fixed by wrapping the require call in a function.

```lua
use {
  'goolord/alpha-nvim',
  requires = {{ "kyazdani42/nvim-web-devicons" }},
  config = function() require('vendion.config.alpha-nvim') end,
}
```

If you are interested in where the `config` key is handled you can find that [here](https://github.com/wbthomason/packer.nvim/blob/98b7212/lua/packer/compile.lua#L262-L278)
